examples/spanner-interleave-subquery/README.md

```diff
@@ -0,0 +1,23 @@
+# spanner-interleave-subquery
+
+This example contains the benchmark code to examine how efficient we can query for Cloud Spanner interleaved tables with subqueries.
+
+## Prerequisite
+
+Create a Cloud Spanner database with [schema.sql](schema.sql).
+
+## How to run the benchmark
+
+Run the following command.
+
+```
+$ go run main.go --project=${PROJECT} --instance=${INSTANCE} --database=${DATABASE} --pattern=${PATTERN} --parallel=${PARALLEL}
+```
+
+For `--pattern` flag please specify the following number.
+
+1. Run benchmark with separated queries for interleaved tables.
+2. Run benchmark with subqueries for interleaved tables.
+3. Insert sample data
+
+You might need to run `--pattern=3` prior to running the benchmark for `--pattern=1` or `--pattern=2` to insert sample data.
```

Comment:
nit can you make pattern 3 pattern 1 in this case so by default if I run down this list in order it always works?

Reply:
suggestion: add a `--pattern=ALL` option to run insert, separate queries, subqueries.

This way you can add an easy e2e test script to:
1) run gcloud / terraform to provision infra
2) go run main.go --pattern=ALL ... (note you can read the project id, instance id, etc. from tf state file or env var)
3) assert zero exit code
4) tear down infra

I'd love to see this in a cloudbuild.yaml so we can run it in the CI for the repo.

Reply:
That makes sense. I replaced pattern 3 with pattern 1.

Regarding `--pattern=ALL`, this benchmark is basically designed to run forever (except inserting sample data) unless user stops the process by `Ctr+C`. So it would be semantically hard to add `--pattern=ALL` at the moment.

Let me consider this again when I will write this blog post in English.